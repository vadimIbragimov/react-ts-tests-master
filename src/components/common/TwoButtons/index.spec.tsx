import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { TwoButtons } from './index';

describe('TwoButtons component', () => {
  it('should slhallow TwoButtons component', () => {
    const component = shallow(<TwoButtons />);
    expect(component).toMatchSnapshot();
  });
  it('should mount TwoButtons component', () => {
    const component = mount(<TwoButtons />);
    expect(component).toMatchSnapshot();
  });
  it('should render TwoButtons component', () => {
    const component = render(<TwoButtons />);
    expect(component).toMatchSnapshot();
  });
  it('should contain text "Кнопка 1"', () => {
    const component = shallow(<TwoButtons />);
    expect(component.find('.one-button').text()).toEqual('Кнопка 1')
  })
});
