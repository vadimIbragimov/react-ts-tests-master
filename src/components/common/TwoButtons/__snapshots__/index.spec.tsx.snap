// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoButtons component should mount TwoButtons component 1`] = `
<TwoButtons>
  <div
    className="blank-div"
  >
    <Button
      className="one-button"
    >
      <Wave>
        <button
          className="ant-btn one-button"
          onClick={[Function]}
          type="button"
        >
          <LoadingIcon
            existIcon={false}
            loading={false}
            prefixCls="ant-btn"
          >
            <CSSMotion
              motionName="ant-btn-loading-icon-motion"
              onAppearActive={[Function]}
              onAppearStart={[Function]}
              onEnterActive={[Function]}
              onEnterStart={[Function]}
              onLeaveActive={[Function]}
              onLeaveStart={[Function]}
              removeOnLeave={true}
              visible={false}
            >
              <DomWrapper />
            </CSSMotion>
          </LoadingIcon>
          <span
            key=".0"
          >
            Кнопка 1
          </span>
        </button>
      </Wave>
    </Button>
    <Button>
      <Wave>
        <button
          className="ant-btn"
          onClick={[Function]}
          type="button"
        >
          <LoadingIcon
            existIcon={false}
            loading={false}
            prefixCls="ant-btn"
          >
            <CSSMotion
              motionName="ant-btn-loading-icon-motion"
              onAppearActive={[Function]}
              onAppearStart={[Function]}
              onEnterActive={[Function]}
              onEnterStart={[Function]}
              onLeaveActive={[Function]}
              onLeaveStart={[Function]}
              removeOnLeave={true}
              visible={false}
            >
              <DomWrapper />
            </CSSMotion>
          </LoadingIcon>
          <span
            key=".0"
          >
            Кнопка 2
          </span>
        </button>
      </Wave>
    </Button>
  </div>
</TwoButtons>
`;

exports[`TwoButtons component should render TwoButtons component 1`] = `
<div
  class="blank-div"
>
  <button
    class="ant-btn one-button"
    type="button"
  >
    <span>
      Кнопка 1
    </span>
  </button>
  <button
    class="ant-btn"
    type="button"
  >
    <span>
      Кнопка 2
    </span>
  </button>
</div>
`;

exports[`TwoButtons component should slhallow TwoButtons component 1`] = `
<div
  className="blank-div"
>
  <Button
    className="one-button"
  >
    Кнопка 1
  </Button>
  <Button>
    Кнопка 2
  </Button>
</div>
`;
